File: sort/selection_sort.js
to_be_sorted = [26, 8, 17, 4, 29, 10, 10]

function selectionSort(arr) {
  let len = arr.length;
  let minIndex, temp;
  for(let i = 0; i < len - 1; i ++) {
    minIndex = i;
    for(j = i + 1; j < len; j ++ ) {
      if(arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }
    temp = arr[i];
    arr[i] = arr[minIndex];
    arr[minIndex] = temp;
  }
  return arr;
}

console.log(selectionSort(to_be_sorted));

File: Go/Sort/selection/selectionSort.go
package main

import "fmt"

func selectionSort(arr *[10]int) {
	var tmp int
	for i := 0; i < len(arr); i++ {
		tmp = i
		for j := i + 1; j < len(arr); j++ {
			if arr[j] < arr[i] {
				tmp = j
			}
		}
		arr[i], arr[tmp] = arr[tmp], arr[i]
	}
}

func main() {
	arr := [10]int{4, 6, 1, 9, 7, 5, 20, 14, 13, 17}
	selectionSort(&arr)
	fmt.Println(arr)
}
